mempool/src/core_mempool/mempool.rs

```diff
@@ -355,6 +355,7 @@ impl Mempool {
                 while skipped.contains(&skipped_txn) {
                     inserted.insert(skipped_txn);
                     result.push(skipped_txn);
+                    skipped.remove(&skipped_txn);
```

Comment:
Unrelated change?

Reply:
Earlier, we weren't removing `skipped_txn` from skipped even after adding the transaction to `inserted`. This is actually resulting some transactions being added to the batch multiple times. 
As it's just a one line change, I combined it with this PR instead of creating a separate PR.